graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/internal/DefaultInputObjectMapper.kt

```diff
@@ -105,50 +105,29 @@ class DefaultInputObjectMapper(private val customInputObjectMapper: InputObjectM
     }
 
     override fun <T> mapToJavaObject(inputMap: Map<String, *>, targetClass: Class<T>): T {
-        if (targetClass == Object::class.java || targetClass == Map::class.java) {
+        if (targetClass.isAssignableFrom(inputMap::class.java)) {
+            @Suppress("UNCHECKED_CAST")
             return inputMap as T
         }
 
         val ctor = ReflectionUtils.accessibleConstructor(targetClass)
-        ReflectionUtils.makeAccessible(ctor)
         val instance = ctor.newInstance()
         var nrOfFieldErrors = 0
-        inputMap.forEach {
-            val declaredField = ReflectionUtils.findField(targetClass, it.key)
-            if (declaredField != null) {
-                val fieldType = getFieldType(declaredField, targetClass)
-                // resolve the field class we will map into, as well as an optional type argument in case such
-                // class is a parameterized type, such as a List.
-                val (fieldClass: Class<*>, fieldArgumentType: Type?) = when (fieldType) {
-                    is ParameterizedType -> fieldType.rawType as Class<*> to fieldType.actualTypeArguments[0]
-                    is Class<*> -> fieldType to null
-                    else -> Class.forName(fieldType.typeName) to null
-                }
-
-                if (it.value is Map<*, *>) {
-                    val mappedValue = if (KotlinDetector.isKotlinType(fieldClass)) {
-                        mapToKotlinObject(it.value as Map<String, *>, fieldClass.kotlin)
-                    } else {
-                        mapToJavaObject(it.value as Map<String, *>, fieldClass)
-                    }
-                    trySetField(declaredField, instance, mappedValue)
-                } else if (it.value is List<*>) {
-                    val newList = convertList(it.value as List<*>, targetClass, fieldClass.kotlin, fieldArgumentType)
-                    if (declaredField.type == Set::class.java) {
-                        trySetField(declaredField, instance, newList.toSet())
-                    } else {
-                        trySetField(declaredField, instance, newList)
-                    }
-                } else if (fieldClass.isEnum) {
```

Comment:
I think we don't have a test covering the enum field case. Should we add one for that as well?

Reply:
There is coverage for Enums already in [InputArgumentTest](https://github.com/Netflix/dgs-framework/blob/534c4a832eeb1808018c8b9ff077a5c16cbcf217/graphql-dgs/src/test/kotlin/com/netflix/graphql/dgs/internal/InputArgumentTest.kt#L1954).

Reply:
I went ahead and added an additional test case for this.